flash/data/process.py

```diff
@@ -101,6 +101,151 @@ class PreprocessState:
 
 
 class Preprocess(Properties, torch.nn.Module):
+    """
+    The :class:`~flash.data.process.Preprocess` is used to encapsulate
+    all the processing data loading logic up to the model.
+
+    It is particularly relevant when you want to provide ready-to-production implementation which works
+    with 4 different stages: ``training``, ``validation``, ``test` and inference(``predicting``).
+
+    The :class:`~flash.data.process.Preprocess` supports the following hooks:
+
+        - ``load_data``: Function to receiving some metadata to generate a Mapping from.
+            Example::
+
+                * Input: Receive a folder path:
+
+                * Action: Walk the folder path and find image paths and their associated labels.
+
+                * Output: Return a list of image paths and their associated labels.
+
+        - ``load_sample``: Function to load a sample from metadata sample.
+            Example::
+
+                * Input: Receive an image path and its label.
+
+                * Action: Load a PIL Image from received image_path.
+
+                * Output: Return the PIL Image and its label.
+
+        - ``pre_tensor_transform``: Preforms transforms on a single data sample object.
```

Comment:
data sample objects sounds redundant to me

```suggestion
        - ``pre_tensor_transform``: Performs transforms on a single data sample.
```

Reply:
Done !